File: code-coaches/medium/text-decompressor/TextDecompressor.java
import java.util.Scanner;

public class TextDecompressor {

    public static String decompressText(String compressedText) {
        StringBuilder decompressedText = new StringBuilder();
        for (int idx = 1; idx < compressedText.length(); idx += 2) {
            char character = compressedText.charAt(idx - 1);
            int repeatCount = Character.getNumericValue(compressedText.charAt(idx));
            decompressedText.append(String.valueOf(character).repeat(repeatCount));
        }

        return decompressedText.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String compressedText = scanner.nextLine();

        System.out.println(decompressText(compressedText));

        scanner.close();
    }

}

File: code-coaches/medium/text-decompressor/TextDecompressor.cs
using System;

public class TextDecompressor
{
    static string DecompressText(string compressedText)
    {
        string decompressedText = "";
        for (int idx = 1; idx < compressedText.Length; idx += 2)
        {
            char character = compressedText[idx - 1];
            int repeatCount = int.Parse(compressedText[idx].ToString());
            decompressedText += new string(character, repeatCount);
        }

        return decompressedText;
    }

    static void Main(string[] args)
    {
        string compressedText = Console.ReadLine();
        Console.WriteLine(DecompressText(compressedText));
    }
}
